doc: Add contribution guideline (#162)
TheThingsIndustries/generic-node-se: CONTRIBUTING.md

### :question: Questions

 If you have questions, please use the [forum](https://www.thethingsnetwork.org/forum/). We have a special [category for The Generic Node](https://www.thethingsnetwork.org/forum/c/nodes/generic-node/88).

### :bulb: New features

 If you have a great idea or think some functionality is missing, we want to know! The only thing you have to do for that is to [create an issue](https://github.com/TheThingsIndustries/generic-node-se/issues) if it doesn't exist yet. Please use the issue template and fill out all sections.

### :beetle: Issues

If you notice that one of the applications is not functioning properly or there may be a bug in the implementation. In this case you should [create an issue](https://github.com/TheThingsIndustries/generic-node-se/issues) if it doesn't exist yet. Please use the issue template and fill out all sections.

### :hammer_and_wrench: Pull requests

If you see an [open issue](https://github.com/TheThingsIndustries/generic-node-se/issues) that you would like to work on, let the other contributors know by commenting in the issue.

### :books: Documentation

> If you see that our documentation is lacking or incorrect, it would be great if you could help us improve it. This will help users and fellow contributors understand how to better develop and use the device. Generic Node documentation [repository](https://github.com/TheThingsIndustries/generic-node-docs) can be viewed at [documentation website](https://thethingsindustries.github.io/generic-node-docs) and you can contribute by following the [documentation contribution guideline](https://github.com/TheThingsIndustries/generic-node-docs/blob/master/CONTRIBUTING.md).

Reviewer: ```suggestion
If you see that our documentation is lacking or incorrect, it would be great if you could help us improve it. This will help users and fellow contributors understand how to better develop and use the device. Generic Node documentation that can be viewed on the [documentation website](https://thethingsindustries.github.io/generic-node-docs) is built from [this repository](https://github.com/TheThingsIndustries/generic-node-docs) and you can contribute to it by following the [documentation contribution guideline](https://github.com/TheThingsIndustries/generic-node-docs/blob/master/CONTRIBUTING.md).
```

## Ready to contribute

- [Fork the repository](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository)

> - Follow our [getting started](https://thethingsindustries.github.io/generic-node-docs/getting-started/) guides to setup your environment.

Reviewer: ```suggestion
- Follow our [getting started](https://thethingsindustries.github.io/generic-node-docs/getting-started/) guides to setup your environment
```

- Create a branch following our [branching guideline](#branching-guideline)

- Commit your changes following our [commit messages guideline](#commit-messages-guideline)

> - Open a pull request

Reviewer: maybe link Github docs on how to create a pull request (like for forking the repo)

* `doc`: documentation
* `sch`: changes to schematic (PIN mapping, HW connection changes,...)
* `pcb`: changes to device PCB and routing
* `hw` : changes to the design (adding new sensor, IC, etc.)
* `dev`: other non-functional development changes, e.g. Makefile, .gitignore, editor config
* `com `: changes to communication ports (UART, I2C, SPI,...)
* `tmr `: changes to timers, counters components
* `compiler`: compiler related changes (attributes, commands,..)
* `lib`:  libraries (additions, changes, improvements, fixes)
* `app`: application level changes (additions, improvements, fixes)
* `all`: changes affecting all code

Changes that affect multiple components can be comma separated.

Good commit messages:

* `doc: Add README with build instructions`
* `dev: Improve CMAKE build with debug and release options`
* `lib,com,app: Fix flash memory read API`

Make sure that commits are scoped to something meaningful and could, potentially, be merged individually.

### Body

The body may contain a more detailed description of the commit, explaining what it changes and why. The "how" is less relevant, as this should be obvious from the diff.

## Legal

> Any contributions related to this repository weather it might be source code improvements or design changes require signing the [Contributor License Agreement (CLA)](https://cla-assistant.io/TheThingsIndustries/generic-node-se).

Reviewer 2: ```suggestion
Any contributions related to this repository whether it might be source code improvements or design changes require signing the [Contributor License Agreement (CLA)](https://cla-assistant.io/TheThingsIndustries/generic-node-se).
```

Author: Done, sorry for that
